addons/point_of_sale/models/pos_session.py

```diff
@@ -2099,6 +2108,117 @@ def get_pos_ui_product_pricelist_item_by_product(self, product_tmpl_ids, product
+        return self.env['ir.config_parameter'].sudo().get_param('point_of_sale.capture_unprocessed_order',
+                                                                'true').lower() == 'true'
+
+    def _handle_order_process_fail(self, order: dict, exception: Exception, draft: bool):
+        if not self._is_capture_system_activated():
+            return
+
+        if draft:
+            # draft will be set when we receive a restaurant order that was not validated yet
+            # if we capture it, it will create tons of (a priori) irrelevant attachments
+            # So we capture only restaurant order that are validated
+            _logger.info("order '%s' was not captured as it is draft", order['data']['name'])
+            return
+
+        # Create records into a dedicated cursor (as the current one will roll back when re-raising the exception)
+        with registry().cursor() as cr:
+            env = api.Environment(
+                cr,
+                SUPERUSER_ID,  # Impersonate odoobot on purpose to bypass permissions on pos.session logging
+                {}
+            )
+            self.with_env(env)._process_order_process_fail(order, exception, self.env.user.id) # purposefully use the original user (and not odoobot)
+
+    def _process_order_process_fail(self, order: dict, exception: Exception, uid: int = False):
+        current_pos_order_data_hash = hash(PoSOrderData.from_data(order['data']))
+        self._get_unprocessed_pos_order_scheduled_activity(
+            order['data']['name'],
+            current_pos_order_data_hash,
+            uid or self.env.user.id,
+            create=True,
+        )
+        self._capture_order_data(order, exception, current_pos_order_data_hash)
+
+    def _shorten_pos_order_data_hash(self, pos_order_data_hash: int):
+        # The bigger the value, the less likely a hash collision will occur
+        # but a bigger value also means a longer name in the attachment
+        return str(pos_order_data_hash)[:6]
+
+    def _get_unprocessed_pos_order_scheduled_activity(self, order_ref: str, pos_order_data_hash: int, assigned_user_id: Optional[int] = None, create: bool = True):
+        if create:
+            assert assigned_user_id, "if in creation mode, user is is needed"
+        pos_module_name = 'point_of_sale'
+        order_ref = order_ref.replace(' ', '_')
+        activity_xid_name = f"activity_pos_unprocessed_{order_ref}_{pos_order_data_hash}"
+        scheduled_activity = self.env.ref(f"{pos_module_name}.{activity_xid_name}", raise_if_not_found=False)
+        if create and not scheduled_activity:
+            scheduled_activity = self.activity_schedule(
+                act_type_xmlid='mail.mail_activity_data_warning',
+                summary=_("PoS order %s can not be processed", order_ref),
+                note=_("The Point of Sale order with the following reference <code>%s #%s</code> "
```

Comment:
Tthanks for bringing up that note is a good next candidate for forcing the markupization.
```suggestion
                note=_("The Point of Sale order with the following reference %s was...
                ", Markup("<code>%s #%s</code>") % (...))
```
(to test)